### PR TITLE
Add outlier_detection to coron3 pipeline step

### PR DESCRIPTION
**Pull request by stsci-hack**
This work updates the outlier_detection step to implement a new parameter, resample_data.  This parameter controls whether resampling is done in outlier_detection or not.  The value of this parameter was then hard-coded in the calwebb_coron3 pipeline step to 'False' since resampling is not needed for coronographic data.  This change in logic should NOT change how outlier_detection works for any other step that calls it since all the original code doing the resampling has simply moved inside if-else blocks.  

The calwebb_coron3 step was also updated to create and then to write out the L2C product generated from outlier_detection.  

These changes were only tested interactively under Python to insure that when the correct inputs are provided, the updated outlier_detection code would work on coronographic data.  However, calwebb_coron3 step as a whole still needs CRDS updates for PSFMASK reference file before it can be tested end-to-end as a pipeline step.

### CODE REVIEW

File: jwst/pipeline/calwebb_coron3.py [hunk 1]
```diff
@@ -123,13 +123,15 @@ def process(self, input):
...
-            # TEMPORAY HACK UNTIL OUTLIER_DETECTION IS VIABLE
-            # Create a dummy level-2c output product
-            log.warning('Creating fake outlier_detection results until step is available')
+            # Create a level-2c output product
+            log.debug('Creating and saving Level-2C result')
             lev2c_name = mk_filename(self.output_dir, target_file, 'calints-'+asn['asn_id'])
             lev2c_model = psf_sub.copy()
+            # Update/replace L2B product DQ array with L2C results
+            for i in range(len(target_models)):
+                psf_sub.dq[i] = target_models[i].dq
```
Comment (hbushouse) [inline jwst/pipeline/calwebb_coron3.py]: In line 131 lev2c_model is created as a copy of psf_sub. So shouldn't it be lev2c_model.dq that gets updated from target_models.dq?

**Comment by hbushouse** [comment]
Applies to #615

**Comment by hbushouse** [comment]
@stscieisenhamer When you have a chance we could use your help in cleaning up and/or streamlining the data flow and saving of products through calwebb_coron3, the same way you've been doing for calwebb_spec3. Of course it actually works as-is, it just isn't "pretty."

File: jwst/pipeline/calwebb_coron3.py (hunk 1 above)
Comment (stsci-hack) [inline jwst/pipeline/calwebb_coron3.py]: Good catch, as you are correct that lev2c_model should be updated.  I will straighten this out shortly...

**Comment by hbushouse** [comment]
@jdavies-st Any comments or suggestions on this before Warren merges?

**Comment by hbushouse** [comment]
Fixes #615. Updates to outlier_detection also fulfill #619 and will be used by #616.

**Pull request merged**